### PR TITLE
Studio:: Add Copy, Paste, Cut, select All and delete entries in menu Edit

### PR DESCRIPTION
**Pull request by medou-boushab**
1- Able to do  Ctrl+X
2- Create a menu: Add Copy, Paste, Cut, select All and delete,

### CODE REVIEW

File: app/scripts/proactive/view/ViewWithProperties.js [hunk 1]
```diff
@@ -62,7 +62,8 @@ define(
 					} else {
 						jobBreadcrumb = $('<li class="active"><span id="breadcrumb-job-name">' + modelInUse.get("Name") + '</span></li>');
 						breadcrumb.append(jobBreadcrumb)
-
+                        //hidden the context-menu
+                        $(".context-menu-canvas").hide();
```
Comment (MaelAudren) [inline app/scripts/proactive/view/ViewWithProperties.js]: Why the indentation is not at the same level ?

File: app/scripts/proactive/view/dom.js [hunk 2]
```diff
@@ -689,18 +689,26 @@ define(
             var studioApp = require('StudioApp');
             studioApp.views.catalogPublishView.publishToCatalog();
         })
-
+        // The aim object of this function is to delete sected tasks
```
Comment (MaelAudren) [inline app/scripts/proactive/view/dom.js]: sected => selected

File: app/scripts/proactive/view/ViewWithProperties.js (hunk 1 above)
Comment (medou-boushab) [reply app/scripts/proactive/view/ViewWithProperties.js]: I don't know why.

Comment (MaelAudren) [reply app/scripts/proactive/view/ViewWithProperties.js]: Can you fix it ?

File: app/scripts/proactive/view/dom.js [hunk 3]
```diff
@@ -1037,43 +1045,90 @@ define(
...
+                })
+                // let the user how he can do past(ctr-v)
+                if(copiedTasks.length > 0){
+                    StudioClient.alert('Copy/Paste', 'Click on the canvas where you want to paste and then do ctrl-V.', 'warning');
+                } else {
+                    StudioClient.alert('Copy/Paste', 'Select at least one task.', 'warning');
+                }
+
+                console.log('positions')
+                console.log(positions)
```
Comment (MaelAudren) [inline app/scripts/proactive/view/dom.js]: Not sure that those log are required and if you thinks that they are required should put them in one line.

File: app/scripts/proactive/view/ViewWithProperties.js (hunk 1 above)
Comment (medou-boushab) [reply app/scripts/proactive/view/ViewWithProperties.js]: I couldn't because I don't have this in intellij

**Comment by fviale** [comment]
In general, very good, but I observe that Cut / Paste does not preserve task dependencies (tried with a replicate block)

**Comment by medou-boushab** [comment]
> In general, very good, but I observe that Cut / Paste does not preserve task dependencies (tried with a replicate block)

I'm trying to fix this.

**Comment by fviale** [comment]
@medou-boushab  Dependencies  are now preserved when we cut/paste, but not replicate script when pasting a replicate block. Other control structures are preserved (If, Loop).
Additionnally, undo does not work properly when we Cut.

**Pull request merged**